### PR TITLE
PAY-5550 Improve Payment App Legacy Functional Tests

### PR DESCRIPTION
**Pull request by sayali-salunkhe**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: model/src/main/java/uk/gov/hmcts/payment/api/service/LoggingCreditAccountPaymentService.java [hunk 1]
```diff
@@ -77,4 +77,10 @@ public List<PaymentFeeLink> search(Date startDate, Date endDate) {
         LOG.info("PaymentFeeLinks found: {}", paymentFeeLinks.size());
         return paymentFeeLinks;
     }
+
+    @Override
+    public void deleteByPaymentReference(String paymentReference) {
+        LOG.info("Deleting payment for reference: {}", paymentReference);
```
Comment (kaushikmaitra) [inline model/src/main/java/uk/gov/hmcts/payment/api/service/LoggingCreditAccountPaymentService.java:83]: Are there only records to be deleted from Payment, none from the other tables?

Comment (sayali-salunkhe) [reply model/src/main/java/uk/gov/hmcts/payment/api/service/LoggingCreditAccountPaymentService.java:83]: All the records from all relational tables are deleted in cascade method

Comment (kaushikmaitra) [reply model/src/main/java/uk/gov/hmcts/payment/api/service/LoggingCreditAccountPaymentService.java:83]: resolved

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java [hunk 2]
```diff
@@ -210,6 +211,17 @@ public ResponseEntity<PaymentDto> retrievePaymentStatus(@PathVariable("paymentRe
...
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "Payment deleted successfully"),
+            @ApiResponse(code = 404, message = "Payment not found for the given reference")
+    })
+    @RequestMapping(value = "/credit-account-payments/{paymentReference}", method = DELETE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteByPaymentReference(@PathVariable("paymentReference") String paymentReference) {
+        creditAccountPaymentService.deleteByPaymentReference(paymentReference);
+    }
+
```
Comment (pats-john) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java:224]: Address the warnings for the Controller Code that is not addressed...

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAAccountsFunctionalTest.java [hunk 3]
```diff
@@ -155,24 +144,16 @@ private final void performOrganisationCreationWithNoAccounts(final String role,f
...
         final String fileContents = String.format(fileContentsTemplate,
             generateRandomString(13, true, false),
             generateRandomString(8, true, false),
             user.getEmail());
-        System.out.println("The value of the File Contents After Templating : " + fileContents);
         Response response = postOrganisation(SERVICE_TOKEN_PAYMENT_APP, testProps.getRefDataApiUrl(), fileContents);
-        System.out.println("The value of the Body" + response.getBody().prettyPrint());
         assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED.value());
         final String organisationIdentifier = response.jsonPath().getString("organisationIdentifier");
-        System.out.println(organisationIdentifier);
```
Comment (pats-john) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAAccountsFunctionalTest.java:170]: Implement HMCTS Logging for the purpose of traceability.....

File: model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCreditAccountPaymentService.java [hunk 4]
```diff
@@ -122,6 +123,16 @@ public List<PaymentFeeLink> search(Date startDate, Date endDate) {
         return paymentFeeLinks;
     }
 
+    @Override
+    @Transactional
+    public void deleteByPaymentReference(String paymentReference) {
+        long records = paymentRespository.deleteByReference(paymentReference);
+        LOG.info("Number of deleted records are: {}", records);
```
Comment (pats-john) [inline model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCreditAccountPaymentService.java:130]: Address all the respective coverage warnings...

**Review by pats-john — CHANGES_REQUESTED** [review]

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 5]
```diff
@@ -79,20 +80,26 @@ public void makeAndRetrievePbaPaymentsByProbate() {
...
-        // Get pba payments by accountNumber
-        PaymentsResponse paymentsResponse = paymentTestService
-                .getPbaPaymentsByAccountNumber(USER_TOKEN, SERVICE_TOKEN, testProps.existingAccountNumber).then()
-                .statusCode(OK.value()).extract().as(PaymentsResponse.class);
+        // Get pba payment by reference
+        PaymentDto paymentsResponse =
+                paymentTestService.getPbaPayment(USER_TOKEN, SERVICE_TOKEN, paymentDto.getReference()).then()
+                        .statusCode(OK.value()).extract().as(PaymentDto.class);
+
+        assertThat(paymentsResponse.getAccountNumber()).isEqualTo(accountNumber);
```
[the remark this answers is not in the document]
Comment (sayali-salunkhe) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:91]: The status and status code are asserted from PaymentDto on Line no. 84

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java (hunk 2 above)
Comment (sayali-salunkhe) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java:224]: Updated code

File: model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCreditAccountPaymentService.java (hunk 4 above)
Comment (sayali-salunkhe) [reply model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCreditAccountPaymentService.java:130]: Updated code

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (sayali-salunkhe) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:91]: Added assertion

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAAccountsFunctionalTest.java (hunk 3 above)
Comment (sayali-salunkhe) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAAccountsFunctionalTest.java:170]: As discussed, will have a new ticket to implement HMCTS Logging across all tests

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java (hunk 2 above)
Comment (pats-john) [reply api/src/main/java/uk/gov/hmcts/payment/api/controllers/CreditAccountPaymentController.java:223]: Resolved.

**Pull request merged**